tests/python/relax/test_op_gradient_numeric.py

```diff
@@ -0,0 +1,408 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import numpy as np
+import pytest
+import tvm
+from tvm import relax
+from tvm.relax.transform import LegalizeOps
+from tvm.testing.utils import check_numerical_grads
+from tvm.ir.op import Op
+from typing import Callable, Union, Tuple, List
+
+
+def relax_check_gradients(
+    op_func: Callable,
+    op_name: str,
+    inputs_numpy: np.array,
+    target: Union[str, tvm.target.Target],
+    dev: tvm._ffi.runtime_ctypes.Device,
+    output_shape: Union[Tuple, List[Tuple]],
+    tuple_input: bool = False,
+    **kwargs,  # attr for operators
+):
+    """Generate module and run it to check numberic gradients."""
+
+    func_name = "main"
+
+    # prepare input
+    def _numpy_to_var(data, var_name):
+        if isinstance(data, list):
+            struct_infos = []
+            for i in range(len(data)):
+                tvm_var = _numpy_to_var(data[i], "")
```

Comment:
Just some easier way
```suggestion
            for _data in data:
                tvm_var = _numpy_to_var(_data, "")
```

Reply:
Ditto for those enumerations below.